components/src/atoms/Tag/index.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react'
+import { css } from 'styled-components'
+import { BORDERS, COLORS } from '../../helix-design-system'
+import { Flex } from '../../primitives'
+import { ALIGN_CENTER, DIRECTION_ROW } from '../../styles'
+import { RESPONSIVENESS, SPACING, TYPOGRAPHY } from '../../ui-style-constants'
+import { Icon } from '../../icons'
+import { StyledText } from '../StyledText'
+
+import type { IconName } from '../../icons'
+import type { StyleProps } from '../../primitives'
+
+export type TagType = 'default' | 'interactive' | 'branded'
+
+interface TagProps extends StyleProps {
+  /** Tag content */
+  text: string
+  /** name constant of the text color and the icon color to display */
+  type: TagType
+  /** iconLocation */
+  iconPosition?: 'left' | 'right'
+  /** Tagicon */
+  iconName?: IconName
+}
+
+const defaultColors = {
+  backgroundColor: `${COLORS.black90}${COLORS.opacity20HexCode}`,
+  color: COLORS.black90,
+}
+
+const TAG_PROPS_BY_TYPE: Record<
+  TagType,
+  {
+    backgroundColor: string
+    color: string
+  }
+> = {
+  default: defaultColors,
+  interactive: defaultColors,
+  branded: {
+    backgroundColor: COLORS.blue50,
+    color: COLORS.white,
+  },
+}
+
+export function Tag(props: TagProps): JSX.Element {
+  const { iconName, type, text, iconPosition, ...styleProps } = props
+
+  const DEFAULT_CONTAINER_STYLE = css`
+    padding: ${SPACING.spacing2} ${SPACING.spacing8};
+    border-radius: ${BORDERS.borderRadius4};
+    @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+      border-radius: ${BORDERS.borderRadius8};
+      padding: ${SPACING.spacing8} ${SPACING.spacing12};
+    }
+  `
+
+  const INTERACTIVE_CONTAINER_STYLE = css`
+    ${DEFAULT_CONTAINER_STYLE}
+    &:hover {
+      background-color: ${COLORS.black90}${COLORS.opacity40HexCode};
+    }
+    &:focus-visible {
+      background-color: ${COLORS.black90}${COLORS.opacity40HexCode};
+      box-shadow: 0 0 0 3px ${COLORS.blue50};
+    }
+  `
```

Comment:
```suggestion
    &:focus-visible {
      box-shadow: 0 0 0 3px ${COLORS.blue50};
      outline: none;
    }
  `
```

![2024-06-18 00_29_37-Library _ Atoms _ Chip - Tag ⋅ Storybook and 10 more pages - Personal - Microsof](https://github.com/Opentrons/opentrons/assets/474225/2e68b5f3-5745-414e-a554-7830335cee5c)



![2024-06-18 00_28_52-Helix Design System – Figma](https://github.com/Opentrons/opentrons/assets/474225/ab875ef0-22a6-4e91-9349-0c986f411036)